Util/src/main/java/io/deephaven/util/JvmIntrospectionContext.java

```diff
@@ -0,0 +1,69 @@
+package io.deephaven.util;
+
+import io.deephaven.util.hotspot.HotSpot;
+
+/**
+ * Utility class to facilitate obtaining data for safepoint pauses count and time between two points in code. A
+ * safepoint pause is a "stop the world, pause all threads" event in the HotSpot JVM. Note full Garbage Collection
+ * pauses are a dominant cause of safepoint pauses, but there are other triggers like: <il>
+ * <li>Deoptimization</li>
+ * <li>Biased lock revocation</li>
+ * <li>Thread dump</li>
+ * <li>Heap inspection< /li>
+ * <li>Class redefinition</li> </il> And others; you can see a full list <a href=
+ * "http://hg.openjdk.java.net/jdk8u/jdk8u/hotspot/file/fc3cd1db10e2/src/share/vm/runtime/vm_operations.hpp#l39"> here
+ * </a>.
+ *
+ */
+
+@SuppressWarnings("restriction")
+public class JvmIntrospectionContext {
+    private static final HotSpot hotspot = HotSpot.instance.orElse(null);
+    private long lastStartPausesCount = -1;
+    private long lastStartPausesTimeMillis = -1;
+    private long lastEndPausesCount;
+    private long lastEndPausesTimeMillis;
+
+    public static boolean hasSafePointData() {
+        return hotspot != null;
+    }
+
+    public JvmIntrospectionContext() {}
+
+    public void startSample() {
+        if (hotspot == null) {
+            return;
+        }
+        lastStartPausesCount = hotspot.getSafepointCount();
+        lastStartPausesTimeMillis = hotspot.getTotalSafepointTimeMillis();
+    }
+
+    /**
+     * Sample garbage collection count and times at the point of call.
```

Comment:
The JavaDocs in this class are inaccurate, maybe they stayed the same while the class evolved?